### PR TITLE
RELATED: RAIL-4439 Documentation on how to use withCaching decorator

### PR DESCRIPTION
**Pull request by plavemar**
JIRA: RAIL-4439

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1146/ : SUCCESS in 3m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/765/ : SUCCESS in 5s

File: docs/18_tips__improve_performance.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+id: improve_performance
+title: Improve Dashboard component's performance
+sidebar_label: Improve Dashboard component's performance
+copyright: (C) 2022 GoodData Corporation
+---
+
+## Caching the backend
```
Comment (no23reason) [inline docs/18_tips__improve_performance.md:8]: Maybe "Caching the backend responses"?

File: docs/18_tips__improve_performance.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+While using the Dashboard component, it's recommended to wrap the backend instance with the `withCaching` decorator for better performance. 
+The decorator needs settings object to be handed over as a parameter to configure where the cache should be applied and to what size should the cache grow. We have provided the recommended caching options (see `RecommendedCachingConfiguration` within the `@gooddata/sdk-backend-base` package.
```
Comment (no23reason) [inline docs/18_tips__improve_performance.md:10]: We should mention that once the max size is reached, the items are removed using LRU policy.

File: docs/18_tips__improve_performance.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+### Caching backend configuration options
```
Comment (kandl) [inline docs/18_tips__improve_performance.md:13]: Maybe we should also add some code snippet or link to live examples, to make it more developer friendly :)

File: docs/18_tips__improve_performance.md (hunk 2 above)
Comment (kandl) [inline docs/18_tips__improve_performance.md:10]: Should we mention Dashboard component only? Caching improves also performance of the embedded insights and all charts ... and even custom executions (thanks to execution cache)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1147/ : SUCCESS in 1m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/766/ : SUCCESS in 5s

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-gate-PR552

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1149/ : SUCCESS in 1m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-fossa-scanning-zuul-el7/253/ : FAILURE in 2m 05s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sonarqube-scan-zuul-gate/148/ : SUCCESS in 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-merge-compliance-check/343/ : SUCCESS in 15s

**Pull request merged**